### PR TITLE
ADRs: 012-016 - Adding Impact Sections 

### PR DESCRIPTION
**Pull request by ghost**
# Description

This PR covers filling in the impact sections for ADRs 012-016 so that it is understood why the decision was made at the time and what the concerns were.

This PR errs on the side of being more verbose than not so that unneeded items can be stripped away during review, so please read carefully. 

**NOTE:** a change to ADR 001 was added to this PR to make linking work better in the IDE.

## Checklist

- [x] I have updated the documentation accordingly

### CODE REVIEW

**Review by GilmoreA6 — APPROVED** [review]

File: adr/013-wrapped-domain-objects.md
Comment (pluckyswan) [inline adr/013-wrapped-domain-objects.md:1]: Is the `FHIR library` the only wrapped domain object currently?

File: adr/016-database.md [hunk 1]
```diff
@@ -15,6 +15,42 @@ Accepted
...
+
+- **Rich Feature Set:** Provides advanced capabilities (e.g., ACID compliance, JSONB, and indexing options) to handle diverse requirements. 
+
+
+- **Community and Ecosystem:** Extensive libraries, tools, and community support minimize development effort and troubleshooting time.
+
+
+### Negative
+
+- **Operational Complexity:** Requires database administration, including backups, updates, and scaling.
```
Comment (jbiskie) [inline adr/016-database.md:34]: Does this need a call out since it could be said for any database?

File: adr/016-database.md [hunk 2]
```diff
@@ -15,6 +15,42 @@ Accepted
...
+
+
+- **Resource Usage:** Postgres can consume significant memory and CPU resources, especially under high load.
+
+
+- **Learning Curve:** Developers or operators unfamiliar with Postgres may require training.
+
+
+### Risks
```
Comment (jbiskie) [inline adr/016-database.md:43]: Similar to my comment above, these would apply to any database.

**Review by jbiskie — APPROVED** [review]

File: adr/012-auth.md [hunk 3]
```diff
@@ -18,13 +18,64 @@ We chose this because it is exactly what ReportStream does and is accepted pract
...
+
+- **Performance Overhead:** Authentication and authorization checks can introduce latency, especially in distributed systems where tokens need to be validated at every request.
+  
+
+- **Misconfiguration:** Incorrect configuration of scopes, permissions, or roles could inadvertently allow unauthorized access to sensitive data.
+
+
+### Risks
+
+- **Vendor Lock-in:** Using third-party AuthN/AuthZ services (e.g., Auth0, AWS Cognito) may lead to vendor lock-in, making it harder to switch providers or implement custom solutions.
```
Comment (basiliskus) [inline adr/012-auth.md:64]: We're not using a third-party service to implement AuthN/AuthZ

**Review by luis-pabon-tf — APPROVED** [review]

File: adr/012-auth.md (hunk 3 above)
Comment (JohnNKing) [reply adr/012-auth.md:64]: > We're not using a third-party service to implement AuthN/AuthZ

+1. I recall we're using a JWT library, but even that should be decoupled via OEA approaches.

File: adr/013-wrapped-domain-objects.md [hunk 4]
```diff
@@ -12,15 +12,52 @@ Accepted.
...
 Our business logic can then use the wrapper Interface without needing to know the actual underlying implementation, and that implementation can be changed without changing the Interface that the business logic relies on.
 
 You can see an example of this in [Order.java](../etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/orders/Order.java) (the interface) and [HapiOrder.java](../etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiOrder.java) (the implementation).
 
+
+## Impact
+
+### Positive
+
+- **Decoupling of Business Logic and Third-Party Libraries:** allows us to decouple our business logic from the specific implementation details of the third-party library. This enhances maintainability and readability.
```
[the remark this answers is not in the document]
Comment (JohnNKing) [reply adr/013-wrapped-domain-objects.md:28]: Meh. There's already a reference to 005-oea.md.

...though I did find an OeA reference included in https://flexion.us/ustc-dawson-case-study/

**Review by JohnNKing — COMMENTED** [review]
Great work!

**Review by jorg3lopez — APPROVED** [review]
Nice work, looks good.

File: adr/013-wrapped-domain-objects.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (tjohnson7021) [reply adr/013-wrapped-domain-objects.md:28]: I added this as an additional reference for transparency. These are reference objects so I don't thing more reasoning is necessarily bad.

**Pull request merged**